### PR TITLE
Session related bugfixes + logging improvements

### PR DESCRIPTION
**Pull request by madjam**
This PR addresses the following issues:
- Ensure server side session state is appropriately cleaned up on session expiration. This helps prevent invalid session state transitions.
- On the client side do not schedule a recurrent keep alive task up front. This helps prevent sending keep alives for sessions the server no longer recognizes.
- Misc logging improvements for better diagnostics

### CODE REVIEW

File: protocol/src/main/java/io/atomix/copycat/protocol/KeepAliveResponse.java [hunk 1]
```diff
@@ -123,7 +123,7 @@ public boolean equals(Object object) {
 
   @Override
   public String toString() {
-    return String.format("%s[status=%s, leader=%s, members=%s]", getClass().getSimpleName(), status, leader, members);
+    return String.format("%s[status=%s, error=%s, leader=%s, members=%s]", getClass().getSimpleName(), status, error, leader, members);
```
Comment (kuujo) [inline protocol/src/main/java/io/atomix/copycat/protocol/KeepAliveResponse.java]: Maybe we should add the `error` to all the other responses too?

Comment (madjam) [inline protocol/src/main/java/io/atomix/copycat/protocol/KeepAliveResponse.java]: I agree. Having this field logged helped a lot. Shall we do that in another PR?

Comment (kuujo) [inline protocol/src/main/java/io/atomix/copycat/protocol/KeepAliveResponse.java]: Sounds good

**Comment by kuujo** [comment]
:+1:
